src/tests/ftest/util/server_utils.py

```diff
@@ -1052,3 +654,271 @@ def get_host(self, rank):
+        Returns:
+            [type]: [description]
+
+        """
+        self._check_information("network", "HostFabrics")
+
+        data = {}
+        try:
+            _info = self.information["network"]["response"]["HostFabrics"]
+            for entry in _info.values():
+                hosts = entry["HostSet"].split(":")[0]
+                data[hosts] = {}
+                if entry["HostFabric"]["Interfaces"]:
+                    for device in entry["HostFabric"]["Interfaces"]:
+                        if device["Device"] not in data[hosts]:
+                            data[hosts][device["Device"]] = []
+                        data[hosts][device["Device"]].append(device["Provider"])
+        except KeyError as error:
+            raise ServerFailed(
+                "ServerInformation: Error obtaining network data: {}".format(
+                    error))
+
+        return data
+
+    def get_storage_capacity(self):
+        """Get the configured SCM and NVMe storage per server engine.
+
+        Only sums up capacities of devices that have been specified in the
+        server configuration file.
+
+        Raises:
+            ServerFailed: if output from the dmg storage scan is missing or
+                not in the expected format
+
+        Returns:
+            dict: a dictionary of each engine's smallest SCM and NVMe storage
+                capacity in bytes, e.g.
+                    {
+                        "scm":  [3183575302144, 6367150604288],
+                        "nvme": [1500312748032, 1500312748032]
+                    }
+
+        """
+        self._check_information("storage", "HostStorage")
+
+        device_capacity = {"nvme": {}, "scm": {}}
+        try:
+            _info = self.information["storage"]["response"]["HostStorage"]
+            for entry in _info.values():
+                # Collect a list of sizes for each NVMe device
+                if entry["storage"]["nvme_devices"]:
+                    for device in entry["storage"]["nvme_devices"]:
+                        if device["pci_addr"] not in device_capacity["nvme"]:
+                            device_capacity["nvme"][device["pci_addr"]] = []
+                        device_capacity["nvme"][device["pci_addr"]].append(0)
+                        for namespace in device["namespaces"]:
+                            device_capacity["nvme"][device["pci_addr"]][-1] += \
+                                namespace["size"]
+
+                # Collect a list of sizes for each SCM device
+                if entry["storage"]["scm_namespaces"]:
+                    for device in entry["storage"]["scm_namespaces"]:
+                        if device["blockdev"] not in device_capacity["scm"]:
+                            device_capacity["scm"][device["blockdev"]] = []
+                        device_capacity["scm"][device["blockdev"]].append(
+                            device["size"])
+
+        except KeyError as error:
+            raise ServerFailed(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword